src/compositemark/common.ts

```diff
@@ -281,12 +282,26 @@ export function compositeMarkOrient<M extends CompositeMark>(
       } else if (xAggregate === compositeMark && yAggregate === compositeMark) {
         throw new Error('Both x and y cannot have aggregate');
       } else {
-        if (isMarkDef(mark) && mark.orient) {
-          return mark.orient;
-        }
+        if (isFieldOrDatumDefForTimeFormat(x)) {
+          // x is temporal
+          if (isFieldOrDatumDefForTimeFormat(y) && isMarkDef(mark) && mark.orient) {
+            // both x and y are temporal and orient is specified
+            return mark.orient;
+          }
 
-        // default orientation = vertical
-        return 'vertical';
+          // x is temporal but y is not
+          return 'vertical';
```

Comment:
If orient is specified, shouldn't we use it?

Reply:
If both x and y are temporal we should follow the orient. But, if only x is temporal, shouldn't the orient be 'vertical'?

Reply:
Well, if the user overrides it, we should respect it or remove the orient but also show a warning. 

Let's write some tests for the different cases to make sure they work.

Reply:
Should we make the user specified orient be the first priority, or just in the temporal case?

Reply:
It should always be the priority but we should remove it and warn the user if we cannot satisfy their request.